feat: remove v1alpha1 subscription (#517)
kyma-project/eventing-manager: docs/user/02-configuration.md

**Spec:**

| Parameter | Type | Description |
| ---- | ----------- | ---- |
| **annotations**  | map\[string\]string | Annotations allows to add annotations to resources. |
| **backend**  | object | Backend defines the active backend used by Eventing. |

> | **backend.&#x200b;config**  | object | Config defines configuration for eventing backend. |

Reviewer: ```suggestion
| **backend.&#x200b;config**  | object | Config defines configuration for the Eventing backend. |
```

| **backend.&#x200b;config.&#x200b;domain**  | string | Domain defines the cluster public domain used to configure the EventMesh Subscriptions

> and their corresponding ApiRules. |

Reviewer: weird line break

> | **backend.&#x200b;config.&#x200b;eventMeshSecret**  | string | EventMeshSecret defines the namespaced name of K8s Secret containing EventMesh credentials. The format of name is "namespace/name". |

Reviewer: ```suggestion
| **backend.&#x200b;config.&#x200b;eventMeshSecret**  | string | EventMeshSecret defines the namespaced name of the Kubernetes Secret containing EventMesh credentials. The format of name is "namespace/name". |
```

| **backend.&#x200b;config.&#x200b;eventTypePrefix**  | string |  |
| **backend.&#x200b;config.&#x200b;natsMaxMsgsPerTopic**  | integer | NATSMaxMsgsPerTopic limits how many messages in the NATS stream to retain per subject. |
| **backend.&#x200b;config.&#x200b;natsStreamMaxSize**  | \{integer or string\} | NATSStreamMaxSize defines the maximum storage size for stream data. |

> | **backend.&#x200b;config.&#x200b;natsStreamReplicas**  | integer | NATSStreamReplicas defines the number of replicas for stream. |

Reviewer: ```suggestion
| **backend.&#x200b;config.&#x200b;natsStreamReplicas**  | integer | NATSStreamReplicas defines the number of replicas for each stream. |
```

just guessing

| **backend.&#x200b;config.&#x200b;natsStreamStorageType**  | string | NATSStreamStorageType defines the storage type for stream data. |
| **backend.&#x200b;type** (required) | string | Type defines which backend to use. The value is either `EventMesh`, or `NATS`. |
| **labels**  | map\[string\]string | Labels allows to add Labels to resources. |
| **logging**  | object | Logging defines the log level for eventing-manager. |
| **logging.&#x200b;logLevel**  | string | LogLevel defines the log level. |
| **publisher**  | object | Publisher defines the configurations for eventing-publisher-proxy. |
| **publisher.&#x200b;replicas**  | object | Replicas defines the scaling min/max for eventing-publisher-proxy. |
| **publisher.&#x200b;replicas.&#x200b;max**  | integer | Max defines maximum number of replicas. |
| **publisher.&#x200b;replicas.&#x200b;min**  | integer | Min defines minimum number of replicas. |
| **publisher.&#x200b;resources**  | object | Resources defines resources for eventing-publisher-proxy. |
| **publisher.&#x200b;resources.&#x200b;claims**  | \[\]object | Claims lists the names of resources, defined in spec.resourceClaims,

> that are used by this container.

Reviewer: weird line break

> This is an alpha field and requires enabling the

Reviewer: also, weird line break here and in the next line

More info: https://kubernetes.io/docs/concepts/configuration/manage-resources-containers/ |

**Status:**

| Parameter | Type | Description |
| ---- | ----------- | ---- |
| **activeBackend** (required) | string |  |
| **conditions**  | \[\]object | Condition contains details for one aspect of the current state of this API Resource.
---
This struct is intended for direct use as an array at the field path .status.conditions.  For example,

	type FooStatus struct{
	    // Represents the observations of a foo's current state.
	    // Known .status.conditions.type are: "Available", "Progressing", and "Degraded"
	    // +patchMergeKey=type
	    // +patchStrategy=merge
	    // +listType=map
	    // +listMapKey=type
	    Conditions []metav1.Condition `json:"conditions,omitempty" patchStrategy:"merge" patchMergeKey:"type" protobuf:"bytes,1,rep,name=conditions"`

	    // other fields
	} |
| **conditions.&#x200b;lastTransitionTime** (required) | string | lastTransitionTime is the last time the condition transitioned from one status to another.
This should be when the underlying condition changed.  If that is not known, then using the time when the API field changed is acceptable. |

> | **conditions.&#x200b;message** (required) | string | message is a human readable message indicating details about the transition.

Reviewer: Why is "message" in lower case? All other rows start in upper case

This may be an empty string. |

> | **conditions.&#x200b;observedGeneration**  | integer | observedGeneration represents the .metadata.generation that the condition was set based upon.

Reviewer: "was set based upon" sounds like one word too much. Pls confirm it's correct; or should this be "was based upon" ?

> For instance, if .metadata.generation is currently 12, but the .status.conditions[x].observedGeneration is 9, the condition is out of date

Reviewer: ```suggestion
For example, if .metadata.generation is currently 12, but the .status.conditions[x].observedGeneration is 9, the condition is out of date
```

with respect to the current state of the instance. |
| **conditions.&#x200b;reason** (required) | string | reason contains a programmatic identifier indicating the reason for the condition's last transition.
Producers of specific condition types may define expected values and meanings for this field,
and whether the values are considered a guaranteed API.
The value should be a CamelCase string.
This field may not be empty. |

> | **conditions.&#x200b;status** (required) | string | status of the condition, one of True, False, Unknown. |

Reviewer: ```suggestion
| **conditions.&#x200b;status** (required) | string | status of the condition, value must be either True, False, or Unknown. |
```

| **conditions.&#x200b;type** (required) | string | type of condition in CamelCase or in foo.example.com/CamelCase.
---

> Many .condition.type values are consistent across resources like Available, but because arbitrary conditions can be

Reviewer: ```suggestion
Many .condition.type values, like Available, are consistent across resources, but because arbitrary conditions can be
```

I think this is what you mean?

useful (see .node.status.conditions), the ability to deconflict is important.
The regex it matches is (dns1123SubdomainFmt/)?(qualifiedNameFmt) |
| **publisherService**  | string |  |
| **specHash** (required) | integer |  |
| **state** (required) | string | Can have one of the following values: Ready, Error, Processing, Warning. Ready state is set
when all the resources are deployed successfully and backend is connected.

> It gets Warning state in case backend is not specified, NATS module is not installed, or EventMesh secret is missing in the cluster.

Reviewer: ```suggestion
It gets Warning state if no backend is specified, NATS module is not installed, or EventMesh secret is missing in the cluster.
```

> Error state is set when there is an error. Processing state is set if recources are being created or changed. |

Reviewer: does this mean "backend not specified, NATs module not installed, Secret missing" are not errors? 
The explanation for "Warning" is very specific, and the explanation for "Error" is extremely vague. Can we improve that?
